encord/objects/coordinates.py

```diff
@@ -158,29 +158,78 @@ class SkeletonCoordinate:
     # `name` and `color` can be removed as they are part of the ontology.
     # The frontend must first be aware of how to merge with ontology.
     name: str
-    color: str
+    color: Optional[str] = None
 
     # `featureHash` and `value` seem to appear when visibility is
     # present. They might not have any meaning. Remove if confirmed that
     # Frontend does not need it.
-    feature_hash: Optional[str] = None
+    featureHash: Optional[str] = None
     value: Optional[str] = None
 
     visibility: Optional[Visibility] = None
 
+    @staticmethod
+    def from_dict(d: dict[str, Any]) -> SkeletonCoordinate:
+        return SkeletonCoordinate(**d)
+
+    def to_dict(self) -> dict:
+        return {
+            "x": self.x,
+            "y": self.y,
+            "name": self.name,
+            "color": self.color,
+            "value": self.value,
+            "featureHash": self.featureHash,
+        }
+
 
 @dataclass(frozen=True)
 class SkeletonCoordinates:
     values: List[SkeletonCoordinate]
 
+    @staticmethod
+    def from_dict(d: dict) -> SkeletonCoordinates:
+        skeleton_dict = d["skeleton"]
+        list_coordinates: List[SkeletonCoordinate] = []
+        sorted_dict_value_tuples = sorted((int(key), value) for key, value in skeleton_dict.items())
+        sorted_dict_values = [item[1] for item in sorted_dict_value_tuples]
+        for coordinate in sorted_dict_values:
+            skeleton_coordinate = SkeletonCoordinate(
+                x=coordinate["x"],
+                y=coordinate["y"],
+                name=coordinate["name"],
+                value=coordinate["value"],
+                color=coordinate["color"],
+                featureHash=coordinate["featureHash"],
+            )
+            list_coordinates.append(skeleton_coordinate)
+        return SkeletonCoordinates(values=list_coordinates)
+
+    def to_dict(self) -> dict:
+        return {i: x.to_dict() for i, x in enumerate(self.values)}
+
+
+@dataclass
+class SkeletonInstance:
+    values: List[SkeletonCoordinate]
+    template: str
+
+    @staticmethod
+    def from_dict(d: dict, skeleton_template: str) -> SkeletonInstance:
+        skeleton_coords = [SkeletonCoordinate(**coord) for coord in d["skeleton"].values()]
+        return SkeletonInstance(skeleton_coords, skeleton_template)
+
+    def to_dict(self) -> dict:
+        return {i: x.to_dict() for i, x in enumerate(self.values)}
+
 
 Coordinates = Union[
     BoundingBoxCoordinates,
     RotatableBoundingBoxCoordinates,
     PointCoordinate,
     PolygonCoordinates,
     PolylineCoordinates,
-    SkeletonCoordinates,
+    SkeletonInstance,
```

Comment:
This is a bad name because 'Instance' is a technical term in the context of labels, meaning 'object entity spanning across frames, with a set of coordinates per frame'.

Also the distinction between `SkeletonCoordinates` and `SkeletonInstance` above is very confusing. They differ only by having the 'template'; what's the intent behind that?

Reply:
Yeah, I wanted to call it SkeletonInstance so as to be clear that "SkeletonInstances" are instantations of templates which is good code but BAD ENCORD code given that Instances are very loaded words and we have ObjectInterfaces in the SDK which the SkeletonInstances fail to implement. Will fix this.